ui/src/app/components/webcomponents/test-step-form-container.component.ts

```diff
@@ -309,8 +309,13 @@ export class TestStepFormContainerComponent extends BaseComponent implements OnI
     }
     testStep.stepDisplayNumber = this.indexPosition(false);
     if(testStep.preRequisiteStepId) {
-      this.testSteps.content.find(testStep => testStep.id == this.testStep.preRequisiteStepId)
-        .stepDisplayNumber = this.indexPosition(true);
+      let preRequisiteStep = this.testSteps.content.find(testStep => testStep.id == this.testStep.preRequisiteStepId)
+        if(preRequisiteStep){
+          preRequisiteStep.stepDisplayNumber = this.indexPosition(true);
+        }
+        else {
+          testStep.preRequisiteStepId=null
+        }
```

Comment:
why this block is required?

Reply:
1) Teststep 1 
2) Teststep 2 [prerequisite step = 1]

when we delete Teststep1 , then Teststep2 still have prerequisitestep = 1 which is not existed.
As Teststep1 is deleted we are assigning prerequisite as null